cmds/insmod/insmod.go

```diff
@@ -31,7 +29,10 @@ func main() {
 	filename := os.Args[1]
 
 	// Everything else is module options
-	options = strings.Join(os.Args[2:], " ")
+	options := " "
```

Comment:
@rminnich: Actually this code is outdated, please check the new one, but the new version also contains this check.

This is very weird... the '2' is the start index, and actually this should be index out of bounds. For example change the '2' iin your google-play code to '23' and it fails. IMO '2' is also out of bounds in a two element slice, valid are only '0' and '1'. Very strange. Is this some documented behavior I've never heard of? Or are we just lucky in this case?

Reply:
First I thought we are lucky because cap() is longer, which it was not. I could imagine (did not check), that the implementation does a if start == end {return //nothing to do}. And in our case end is len(). Just guessing, but that would at least explain why it does not fail.

Reply:
it's the documented behavior. It's  the so-called 'half-open' range. In a two element array, x[2:] is valid and an empty slice.

Reply:
@rminnich : I already pushed the version without the check, but out of curiosity, where would that be documented? Yes, I know what an half-open range is, but I can't find any reference that start == len() would be guaranteed to be okay. Any references?

Reply:
Okay, here we go:
"For arrays or strings, the indices are in range if 0 <= low <= high <= len(a), otherwise they are out of range. For slices, the upper index bound is the slice capacity cap(a) rather than the length." Indeed all "<=". Thanks, learned something new :-)